File: recursive-count-master/ruby/recursive_count.rb
def recursive_count(number = 0)
  if number < 10
    puts number # output the number
    recursive_count(number + 1) # recursively call the function with the next number
  end
end

 #{The method takes an optional parameter number with a default value of 0. It then checks if the number is less than 10. If it is, it uses puts to output the number and recursively calls recursive_count with the next number (number + 1).
 
if __FILE__ == $PROGRAM_NAME
  recursive_count
end

# OPTIONAL
# Please add your pseudocode to this file
# And a written explanation of your solution

File: recursive-count-master/javascript/recursive_count.js
function recursiveCount(num = 0) {
  // type your code here
}

if (require.main === module) {
  recursiveCount();
}

module.exports = recursiveCount;

// OPTIONAL
// Please add your pseudocode to this file
// And a written explanation of your solution

//const recursiveCount = require('../recursive_count');

//test('logs numbers 0 to 9', () => {
 // console.log = jest.fn();
  //recursiveCount();

  //for (let i = 0; i < 10; ++i) {
  //  expect(console.log).toHaveBeenCalledWith(i);
 // }
//});

//test('returns undefined', () => {
 // expect(recursiveCount()).toBe(undefined);
//});


function recursive_count( i = 0) {
  if ( i < 10) {
    console.lof(i);
    recursiveCount( i = 1);
  }
}

recursiveCount();





const recursiveCount = require('../recursive_count');

test('logs numbers 0 to 9', () => {
  console.log = jest.fn();
  recursiveCount();

  for (let i = 0; i < 10; ++i) {
    expect(console.log).toHaveBeenCalledWith(i);
  }
});

test('returns undefined', () => {
  expect(recursiveCount()).toBe(undefined);
});




